src/trio/_core/_parking_lot.py

```diff
@@ -86,6 +87,28 @@
     from ._run import Task
 
 
+GLOBAL_PARKING_LOT_BREAKER: dict[Task, list[ParkingLot]] = {}
+
+
+def add_parking_lot_breaker(task: Task, lot: ParkingLot) -> None:
+    """Register a task as a breaker for a lot. This means that if the task exits without
+    having unparked from the lot, then the lot will break and raise an error for all tasks
+    parked in the lot, as well as any future task that attempt to park in it."""
+    if task not in GLOBAL_PARKING_LOT_BREAKER:
+        GLOBAL_PARKING_LOT_BREAKER[task] = [lot]
+    else:
+        GLOBAL_PARKING_LOT_BREAKER[task].append(lot)
+
+
+def remove_parking_lot_breaker(task: Task, lot: ParkingLot) -> None:
+    """Deregister a task as a breaker for a lot. See :func:`add_parking_lot_breaker`."""
+    if task not in GLOBAL_PARKING_LOT_BREAKER:
+        raise RuntimeError(
+            "Attempted to remove parking lot breaker for task that is not registered as a breaker",
+        )
+    GLOBAL_PARKING_LOT_BREAKER[task].remove(lot)
```

Comment:
Instead of directly checking each time, this could just catch the `KeyError` (then reraise and suppress), so it has zero overhead in the happy case. Might also want to catch `ValueError` from `remove()` failing.

Reply:
Fixed

and tbh it's probably better not to distinguish between `KeyError` and `ValueError`. Which one you'll get will depend on if the task has previously parked any lots or not, not much about the current state.